<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualization • chevreulPlot</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Visualization">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">chevreulPlot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.27</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/chevreulPlot.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/visualization.html">Visualization</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/cobriniklab/chevreulPlot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualization</h1>
                        <h4 data-toc-skip class="author">Kevin
Stachelek</h4>
            <address class="author_afil">
      University of Southern
California<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.stachelek@gmail.com" class="email">kevin.stachelek@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Bhavana
Bhat</h4>
            <address class="author_afil">
      University of Southern
California<br><a class="author_email" href="mailto:#"></a><a href="mailto:bbhat@usc.edu" class="email">bbhat@usc.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">15 January 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cobriniklab/chevreulPlot/blob/master/vignettes/visualization.Rmd" class="external-link"><code>vignettes/visualization.Rmd</code></a></small>
      <div class="d-none name"><code>visualization.Rmd</code></div>
    </div>

    
    
<p>This article demonstrates the data visualization tools in
chevreulPlot. We’ll introduce included functions, their usage, and
resulting plots</p>
<p>First step is to load chevreulPlot package and all other packages
required</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cobriniklab/chevreulPlot" class="external-link">chevreulPlot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MarioniLab/scran/" class="external-link">scran</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lazappi/clustree" class="external-link">clustree</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cobriniklab/chevreuldata" class="external-link">chevreuldata</a></span><span class="op">)</span></span>
<span><span class="va">chevreul_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/chevreuldata/man/human_gene_transcript_sce.html" class="external-link">human_gene_transcript_sce</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>The different plotting functions within chevreulPlot allows for
visualization of data, these plots can be customized for interactive or
non-interactive display.</p>
<div class="section level2">
<h2 id="plot-expression">Plot expression<a class="anchor" aria-label="anchor" href="#plot-expression"></a>
</h2>
<p>Expression of a feature (genes or transcripts) can be plotted on a
given embedding resulting in an interactive feature plot.</p>
<p>When plotting only one feature, output is identical to
<code>SingleCellExperiment::FeaturePlot</code></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_feature_on_embedding.html">plot_feature_on_embedding</a></span><span class="op">(</span><span class="va">chevreul_sce</span>,</span>
<span>    embedding <span class="op">=</span> <span class="st">"UMAP"</span>,</span>
<span>    features <span class="op">=</span> <span class="st">"NRL"</span>, return_plotly <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-1-1.png" width="100%"></p>
<p>An interactive output plot can be generated by specifying
<code>return_plotly = TRUE</code> which uses <code>ggplotly</code>
allowing identification of individual cells for further
investigation.</p>
<div class="section level3">
<h3 id="plot-read-count-or-other-qc-measurements">Plot read count or other QC measurements<a class="anchor" aria-label="anchor" href="#plot-read-count-or-other-qc-measurements"></a>
</h3>
<p>The <code>plot_colData_histogram</code> function displays a histogram
of cell read counts colored according to a categorical variable using
the argument <code>fill_by</code>. Here we can see that read counts for
this dataset are distinctly different depending on the sequencing
batch</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_colData_histogram.html">plot_colData_histogram</a></span><span class="op">(</span><span class="va">chevreul_sce</span>,</span>
<span>    group_by <span class="op">=</span> <span class="st">"nCount_Gene"</span>,</span>
<span>    fill_by <span class="op">=</span> <span class="st">"batch"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-2-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="plot-metadata-variable">Plot metadata variable<a class="anchor" aria-label="anchor" href="#plot-metadata-variable"></a>
</h3>
<p>Make an interactive scatter plot of a metadata variable, where each
point in the plot represents a cell whose position on the plot is given
by the cell embedding determined by the dimensional reduction technique
by default, “UMAP”. The group argument specifies the colData variable by
which to group the cells by, by default, “batch”.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_colData_on_embedding.html">plot_colData_on_embedding</a></span><span class="op">(</span><span class="va">chevreul_sce</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span>,</span>
<span>    embedding <span class="op">=</span> <span class="st">"UMAP"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-3-1.png" width="100%"></p>
<p>This function utilizes a SingleCellExperiment function,
<code>DimPlot()</code>, as sub function which produces the dimensional
reduction plot. The interactive parameter, <code>return_plotly</code>,
in plot_colData_on_embedding when set to TRUE will convert the plot into
an interactive plot using ggplotly function from R’s plotly package</p>
</div>
<div class="section level3">
<h3 id="plot-cluster-marker-genes">Plot cluster marker genes<a class="anchor" aria-label="anchor" href="#plot-cluster-marker-genes"></a>
</h3>
<p>Marker genes of louvain clusters or additional experimental metadata
can be plotted using <code>plot_marker_features</code>. This allows
visualization of n marker features grouped by the metadata of interest.
Marker genes are identified using wilcoxon rank-sum test as implemented
in <code>presto</code>. In the resulting dot plot the size of the dot
corresponds to the percentage of cells expressing the feature in each
cluster and the color represents the average expression level of the
feature.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_marker_features.html">plot_marker_features</a></span><span class="op">(</span><span class="va">chevreul_sce</span>,</span>
<span>    group_by <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span>,</span>
<span>    marker_method <span class="op">=</span> <span class="st">"wilcox"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-4-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="plotting-transcript-composition">Plotting transcript composition<a class="anchor" aria-label="anchor" href="#plotting-transcript-composition"></a>
</h3>
<p><code><a href="../reference/plot_transcript_composition.html">plot_transcript_composition()</a></code> plots the proportion of
reads of a given gene map to each transcript. The gene of interest is
specified by the argument ‘gene_symbol’.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-5-1.png" width="100%"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] chevreuldata_0.99.25        ExperimentHub_2.14.0       </span></span>
<span><span class="co">#&gt;  [3] AnnotationHub_3.14.0        BiocFileCache_2.14.0       </span></span>
<span><span class="co">#&gt;  [5] dbplyr_2.5.0                patchwork_1.3.0            </span></span>
<span><span class="co">#&gt;  [7] clustree_0.5.1              ggraph_2.2.1               </span></span>
<span><span class="co">#&gt;  [9] scran_1.34.0                chevreulPlot_0.99.27       </span></span>
<span><span class="co">#&gt; [11] chevreulProcess_0.99.23     scater_1.34.0              </span></span>
<span><span class="co">#&gt; [13] ggplot2_3.5.1               scuttle_1.16.0             </span></span>
<span><span class="co">#&gt; [15] SingleCellExperiment_1.28.1 SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt; [17] Biobase_2.66.0              GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt; [19] GenomeInfoDb_1.42.1         IRanges_2.40.1             </span></span>
<span><span class="co">#&gt; [21] S4Vectors_0.44.0            BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt; [23] MatrixGenerics_1.18.1       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [25] BiocStyle_2.34.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] batchelor_1.22.0          BiocIO_1.16.0            </span></span>
<span><span class="co">#&gt;   [3] filelock_1.0.3            bitops_1.0-9             </span></span>
<span><span class="co">#&gt;   [5] tibble_3.2.1              polyclip_1.10-7          </span></span>
<span><span class="co">#&gt;   [7] XML_3.99-0.18             lifecycle_1.0.4          </span></span>
<span><span class="co">#&gt;   [9] edgeR_4.4.1               doParallel_1.0.17        </span></span>
<span><span class="co">#&gt;  [11] lattice_0.22-6            ensembldb_2.30.0         </span></span>
<span><span class="co">#&gt;  [13] MASS_7.3-61               magrittr_2.0.3           </span></span>
<span><span class="co">#&gt;  [15] limma_3.62.2              plotly_4.10.4            </span></span>
<span><span class="co">#&gt;  [17] sass_0.4.9                rmarkdown_2.29           </span></span>
<span><span class="co">#&gt;  [19] jquerylib_0.1.4           yaml_2.3.10              </span></span>
<span><span class="co">#&gt;  [21] metapod_1.14.0            cowplot_1.1.3            </span></span>
<span><span class="co">#&gt;  [23] DBI_1.2.3                 RColorBrewer_1.1-3       </span></span>
<span><span class="co">#&gt;  [25] ResidualMatrix_1.16.0     abind_1.4-8              </span></span>
<span><span class="co">#&gt;  [27] zlibbioc_1.52.0           purrr_1.0.2              </span></span>
<span><span class="co">#&gt;  [29] AnnotationFilter_1.30.0   RCurl_1.98-1.16          </span></span>
<span><span class="co">#&gt;  [31] rappdirs_0.3.3            tweenr_2.0.3             </span></span>
<span><span class="co">#&gt;  [33] circlize_0.4.16           GenomeInfoDbData_1.2.13  </span></span>
<span><span class="co">#&gt;  [35] ggrepel_0.9.6             irlba_2.3.5.1            </span></span>
<span><span class="co">#&gt;  [37] megadepth_1.16.0          cmdfun_1.0.2             </span></span>
<span><span class="co">#&gt;  [39] dqrng_0.4.1               pkgdown_2.1.1            </span></span>
<span><span class="co">#&gt;  [41] DelayedMatrixStats_1.28.1 codetools_0.2-20         </span></span>
<span><span class="co">#&gt;  [43] DelayedArray_0.32.0       ggforce_0.4.2            </span></span>
<span><span class="co">#&gt;  [45] tidyselect_1.2.1          shape_1.4.6.1            </span></span>
<span><span class="co">#&gt;  [47] UCSC.utils_1.2.0          farver_2.1.2             </span></span>
<span><span class="co">#&gt;  [49] wiggleplotr_1.30.0        ScaledMatrix_1.14.0      </span></span>
<span><span class="co">#&gt;  [51] viridis_0.6.5             GenomicAlignments_1.42.0 </span></span>
<span><span class="co">#&gt;  [53] jsonlite_1.8.9            GetoptLong_1.0.5         </span></span>
<span><span class="co">#&gt;  [55] BiocNeighbors_2.0.1       tidygraph_1.3.1          </span></span>
<span><span class="co">#&gt;  [57] iterators_1.0.14          systemfonts_1.1.0        </span></span>
<span><span class="co">#&gt;  [59] foreach_1.5.2             tools_4.4.2              </span></span>
<span><span class="co">#&gt;  [61] ragg_1.3.3                Rcpp_1.0.14              </span></span>
<span><span class="co">#&gt;  [63] glue_1.8.0                gridExtra_2.3            </span></span>
<span><span class="co">#&gt;  [65] SparseArray_1.6.0         xfun_0.50                </span></span>
<span><span class="co">#&gt;  [67] dplyr_1.1.4               withr_3.0.2              </span></span>
<span><span class="co">#&gt;  [69] BiocManager_1.30.25       fastmap_1.2.0            </span></span>
<span><span class="co">#&gt;  [71] bluster_1.16.0            digest_0.6.37            </span></span>
<span><span class="co">#&gt;  [73] rsvd_1.0.5                mime_0.12                </span></span>
<span><span class="co">#&gt;  [75] R6_2.5.1                  textshaping_0.4.1        </span></span>
<span><span class="co">#&gt;  [77] colorspace_2.1-1          RSQLite_2.3.9            </span></span>
<span><span class="co">#&gt;  [79] utf8_1.2.4                tidyr_1.3.1              </span></span>
<span><span class="co">#&gt;  [81] generics_0.1.3            data.table_1.16.4        </span></span>
<span><span class="co">#&gt;  [83] rtracklayer_1.66.0        graphlayouts_1.2.1       </span></span>
<span><span class="co">#&gt;  [85] httr_1.4.7                htmlwidgets_1.6.4        </span></span>
<span><span class="co">#&gt;  [87] S4Arrays_1.6.0            pkgconfig_2.0.3          </span></span>
<span><span class="co">#&gt;  [89] gtable_0.3.6              blob_1.2.4               </span></span>
<span><span class="co">#&gt;  [91] ComplexHeatmap_2.22.0     XVector_0.46.0           </span></span>
<span><span class="co">#&gt;  [93] htmltools_0.5.8.1         bookdown_0.42            </span></span>
<span><span class="co">#&gt;  [95] ProtGenerics_1.38.0       clue_0.3-66              </span></span>
<span><span class="co">#&gt;  [97] scales_1.3.0              png_0.1-8                </span></span>
<span><span class="co">#&gt;  [99] knitr_1.49                tzdb_0.4.0               </span></span>
<span><span class="co">#&gt; [101] rjson_0.2.23              curl_6.1.0               </span></span>
<span><span class="co">#&gt; [103] cachem_1.1.0              GlobalOptions_0.1.2      </span></span>
<span><span class="co">#&gt; [105] stringr_1.5.1             BiocVersion_3.20.0       </span></span>
<span><span class="co">#&gt; [107] parallel_4.4.2            vipor_0.4.7              </span></span>
<span><span class="co">#&gt; [109] AnnotationDbi_1.68.0      restfulr_0.0.15          </span></span>
<span><span class="co">#&gt; [111] desc_1.4.3                pillar_1.10.1            </span></span>
<span><span class="co">#&gt; [113] grid_4.4.2                vctrs_0.6.5              </span></span>
<span><span class="co">#&gt; [115] BiocSingular_1.22.0       EnsDb.Hsapiens.v86_2.99.0</span></span>
<span><span class="co">#&gt; [117] beachmat_2.22.0           cluster_2.1.6            </span></span>
<span><span class="co">#&gt; [119] beeswarm_0.4.0            evaluate_1.0.3           </span></span>
<span><span class="co">#&gt; [121] readr_2.1.5               GenomicFeatures_1.58.0   </span></span>
<span><span class="co">#&gt; [123] cli_3.6.3                 locfit_1.5-9.10          </span></span>
<span><span class="co">#&gt; [125] compiler_4.4.2            Rsamtools_2.22.0         </span></span>
<span><span class="co">#&gt; [127] rlang_1.1.4               crayon_1.5.3             </span></span>
<span><span class="co">#&gt; [129] labeling_0.4.3            forcats_1.0.0            </span></span>
<span><span class="co">#&gt; [131] fs_1.6.5                  ggbeeswarm_0.7.2         </span></span>
<span><span class="co">#&gt; [133] stringi_1.8.4             viridisLite_0.4.2        </span></span>
<span><span class="co">#&gt; [135] BiocParallel_1.40.0       munsell_0.5.1            </span></span>
<span><span class="co">#&gt; [137] Biostrings_2.74.1         lazyeval_0.2.2           </span></span>
<span><span class="co">#&gt; [139] Matrix_1.7-1              hms_1.1.3                </span></span>
<span><span class="co">#&gt; [141] sparseMatrixStats_1.18.0  bit64_4.5.2              </span></span>
<span><span class="co">#&gt; [143] KEGGREST_1.46.0           statmod_1.5.0            </span></span>
<span><span class="co">#&gt; [145] igraph_2.1.3              memoise_2.0.1            </span></span>
<span><span class="co">#&gt; [147] bslib_0.8.0               bit_4.5.0.1</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kevin Stachelek, Bhavana Bhat.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
